src/common/helpers.ts

```diff
@@ -66,3 +66,26 @@ export function hasAdditionalProperties(obj: ObjectSchema): DictionarySchema | u
   }
   return undefined;
 }
+
+export function isScalar(schema: Schema): boolean {
+  switch (schema.type) {
+    case SchemaType.Array:
+      return isScalar((<ArraySchema>schema).elementType);
+    case SchemaType.Binary:
```

Comment:
I think we can remove this one since we don't create a response envelope for binary responses.  The rest look good.

Reply:
Removed.